chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -1328,32 +1326,49 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
 
     chunks = loadedChunks;
     camera.setWorldSize(1 << worldOctree.getDepth());
-    try (TaskTracker.Task task = taskTracker.task("Building BVH")) {
+    try (TaskTracker.Task task = taskTracker.task("(5/6) Building world BVH")) {
       buildBvh(task);
+    }
+    try (TaskTracker.Task task = taskTracker.task("(6/6) Building actor BVH")) {
       buildActorBvh(task);
     }
     Log.info(String.format("Loaded %d chunks", numChunks));
   }
 
   private void buildBvh(TaskTracker.Task task) {
-    task.update("Building world BVH", 1, 0);
-    final List<Primitive> primitives = new LinkedList<>();
+    double entityScaler = 500.0 / entities.size();
+    int done = 0;
+    task.update(1000, 0);
 
+    final List<Primitive> primitives = new LinkedList<>();
     Vector3 worldOffset = new Vector3(-origin.x, -origin.y, -origin.z);
     for (Entity entity : entities) {
       primitives.addAll(entity.primitives(worldOffset));
+
+      done++;
+      task.updateInterval((int) (done * entityScaler), 1);
     }
-    bvh = new BVH(primitives, task);
+
+    double primitiveScaler = 500.0 / primitives.size();
+    bvh = new BVH(primitives, i -> task.updateInterval((int) (i * primitiveScaler) + 500, 1));
   }
 
   private void buildActorBvh(TaskTracker.Task task) {
-    task.update("Building actor BVH", 1, 0);
+    double entityScaler = 500.0 / entities.size();
+    int done = 0;
+    task.update(1000, 0);
```

Comment:
What's up with the scaling? Why not just `task.update(entities.size(), 0);`?

Reply:
The scaling is since we don't actually know how many elements there are going to be. Ideally, it should be `entities.size() + primitives.size()`, but we don't know the primitive length until it's halfway done. In addition, loading primitives takes about half the time which a pure count based approach wouldn't reflect.

Reply:
Oh, I see… then I guess 1000 and 500 are used because they are "big enough" to show some actual progress? Should we make`task.update` accept doubles instead in the future?

Reply:
An `updatePercent` might be useful. I'll have to see when (if) I refactor all the other usages of TaskTracker.

Reply:
I don't expect you to do that :upside_down_face: Maybe `task.subtask(scale)` which returns a new `Task` that will update the original task whenever it is updated? :thinking: Can be done at a later point though, I'll merge this now. The progress bar improved a lot already thanks to your work :pray: